### PR TITLE
first example for java (test with maven)

### PR DESCRIPTION
**Pull request by tyrcho**
I'm not sure about the strategy with docker here.

I didn't feel the need to use a Dockerfile like other examples.

I'm not sure if we want to mount the `.m2/repository` volume like I did. Without this volume the build can quickly get quite slow.

Also I mounted separately `src` and `pom.xml` rather than simply `$(pwd)` to avoid the root user from the container writing stuff in `target` which is a pain to cleanup afterwards.

### CODE REVIEW

**Comment by tsunammis** [comment]
@tyrcho thank you for your contribution!

```
I didn't feel the need to use a Dockerfile like other examples.
```
It's not mandatory to use a Dockerfile. Most of the time, we create one on our part, as we use that technology in production.

```
I'm not sure if we want to mount the .m2/repository volume like I did. Without this volume the build can quickly get quite slow.
```
The idea of Gazr is to provide a universal way to launch common task without any dependencies on the developer laptop. Thus, regarding your example, we consider that the developer doesn't have "maven" on its laptop. According to me, we have to remove the mount of ".m2/repository"

File: examples/java/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+target/
```
Comment (tsunammis) [inline examples/java/.gitignore]: As your example provides a "test-unit" code sample, could you rename the folder name from `java` to `test_java_junit`?

**Review by tsunammis — CHANGES_REQUESTED** [review]

**Comment by tyrcho** [comment]
I did the requested changes.

People using maven on the host, and with company proxy, might find useful to add this line to the makefile:

	  -v ~/.m2/settings.xml:/root/.m2/settings.xml \

I'm wondering how practical *gazr* would be for java developers ... most projects tend to "download the universe" via maven. I guess non java developers depending on a java µservice would only start it once per day so start time (expect a few minutes) would not be that much an issue.

**Review by tsunammis — APPROVED** [review]

**Comment by tsunammis** [comment]
@tyrcho I'm not an expert in Java, so, I completely trust you about the java example ;-)
Could you explain the reason(s) why it will be not practical for Java developers to use Gazr?
As I don't develop with java very often, I'm not the suited guy to provide relevant examples.

**Comment by tyrcho** [comment]
@tsunammis I was concerned about how slow the builds would be with maven, but after discussion with a colleague we found a solution by mounting the appropriate folder so it is only slow the first time. This is the purpose of PR #6 .

**Pull request merged**